flink-spark-bundle/src/test/java/org/apache/iceberg/flinkspark/TestFlinkSpark.java

```diff
@@ -0,0 +1,348 @@
+        FlinkSink.Builder builder =
+                FlinkSink.forRow(dataStream, SimpleDataUtil.FLINK_SCHEMA).table(table);
+
+        // Use schema identifier field IDs as equality field id list by default
+        assertThat(builder.checkAndGetEqualityFieldIds())
+                .containsExactlyInAnyOrderElementsOf(table.schema().identifierFieldIds());
+
+        // Use user-provided equality field column as equality field id list
+        builder.equalityFieldColumns(Lists.newArrayList("id"));
+        assertThat(builder.checkAndGetEqualityFieldIds())
+                .containsExactlyInAnyOrder(table.schema().findField("id").fieldId());
+
+        builder.equalityFieldColumns(Lists.newArrayList("type"));
+        assertThat(builder.checkAndGetEqualityFieldIds())
+                .containsExactlyInAnyOrder(table.schema().findField("type").fieldId());
+    }
+
+    @TestTemplate
+    public void testEqualityDeleteWritesOnSpark() throws Exception {
+        // Step 1: Write initial data using Flink
+        DataStream<Row> dataStream = env.fromCollection(
+                ImmutableList.of(
+                        row("+I", 1, "value1"),
+                        row("+I", 2, "value2"),
+                        row("+I", 3, "value3")
+                )
+        );
+
+
+        FlinkSink.forRow(dataStream, SimpleDataUtil.FLINK_SCHEMA)
+                .tableLoader(tableLoader)
+                .writeParallelism(parallelism)
+                .append();
+
+        // Execute the Flink job to write initial data
+        env.execute("Write Initial Data");
+
+        // Step 2: Apply equality deletes using Flink
+
+        DataStream<Row> deleteStream = env.fromCollection(
+                ImmutableList.of(
+                        row("-D", 1, "value1"),  // Equality delete row with id=1
+                        row("-D", 2, "value2")   // Equality delete row with id=2
+                )
+        );
+
+        FlinkSink.forRow(deleteStream, SimpleDataUtil.FLINK_SCHEMA)
+                .tableLoader(tableLoader)
+                .equalityFieldColumns(Lists.newArrayList("id", "data"))
+                .writeParallelism(parallelism)
+                .upsert(true)  // Enable UPSERT mode for equality deletes
+                .append();
+
+        // Execute the Flink job to apply equality deletes
+        env.execute("Apply Equality Deletes");
+
+        // Step 3: Use Spark to read the table and verify that equality deletes were applied correctly
+
+        // Initialize SparkSession
+        SparkSession spark = SparkSession.builder()
+                .appName("IcebergSparkRead")
+                .master("local[*]")
+                .config("spark.sql.catalog.hadoop_catalog", "org.apache.iceberg.spark.SparkCatalog")
+                .config("spark.sql.catalog.hadoop_catalog.type", "hadoop")
+                .config("spark.sql.catalog.myCatalog.warehouse", "file:///path/to/warehouse")
```

Comment:
We should set the same warehouse as the Flink catalog to ensure we are using the same tables.

```
spark = SparkSession.builder()
        .appName("iceberg-spark")
        .master("local[*]")
        .config("spark.sql.catalog.hadoop_catalog", "org.apache.iceberg.spark.SparkCatalog")
        .config("spark.sql.catalog.hadoop_catalog.type", "hadoop")
        .config("spark.sql.catalog.hadoop_catalog.warehouse", CATALOG_EXTENSION.warehouse())
        .getOrCreate();

```